index.js

```diff
@@ -279,6 +279,9 @@
   var l = $.TypeVariable('l');
   var r = $.TypeVariable('r');
 
+  //  Predicate :: Type
+  var Predicate = $._UnaryFunction($.Any, $.Boolean);
```

Comment:
Sounds good to me, since in each case the predicate must accept any value. For a function such as `filter`, though, we'd want `a -> Boolean` rather than `Any -> Boolean`.